setup.py

```diff
@@ -47,10 +47,9 @@ def check_cuda_torch_binary_vs_bare_metal(cuda_dir):
     print(raw_output + "from " + cuda_dir + "/bin\n")
 
     if (bare_metal_major != torch_binary_major) or (bare_metal_minor != torch_binary_minor):
```

Comment:
Pytorch supports Cuda 9.1. Not sure that minor versions must be also compared.

Reply:
True, and 9.2 as well.  Nvidia also ships Docker containers where we build Pytorch from source with cuda 10.1.  In practice, I do find that sometimes minor version mismatches can cause errors.

Reply:
In my case it did not. I had to edit the file and remove this check.

Reply:
I think it's good to keep this check for safety.  I will note what you said in the error message, telling people they can comment it out if necessary, but at their own risk.

Reply:
Maybe the judge condition(i mean the if)can be other specific ones not just use the cuda version.

Reply:
@proszx I'm not sure what you mean by "other specific ones". Could you explain it a bit?

Reply:
That's helps me a lot. Thanks.

Reply:
This is outdated, here is an updated diff:

```
diff --git a/setup.py b/setup.py
index e3063be..ed88abd 100644
--- a/setup.py
+++ b/setup.py
@@ -30,6 +30,11 @@ def check_cuda_torch_binary_vs_bare_metal(cuda_dir):
     print(raw_output + "from " + cuda_dir + "/bin\n")

     if (bare_metal_version != torch_binary_version):
+
+        # allow minor diffs
+        if bare_metal_version.major == torch_binary_version.major and bare_metal_version.minor != torch_binary_version.minor:
+            return
+
         raise RuntimeError(
             "Cuda extensions are being compiled with a version of Cuda that does "
             "not match the version used to compile Pytorch binaries.  "
```

@ptrblck, perhaps this could be added to `setup.py` in commented out fashion so it's trivial for the user to activate? This is a safer approach than just outright `return`, since it'll check that the major version matches.

Reply:
or even better, let's have an env var that activates this so that the error could include, run:

```
CUDA_MINOR_VERSION_MISMATCH_OK=1 python setup.py ...
```

so then the user doesn't have to change the source code at all.

Reply:
I installed CUDA 10.2 and pytorch 1.7.1 in my conda virtual env, and Cuda compilation tools is 11.1, it didn't work for me.

Reply:
torch.__version__  = 2.3.0+cu121

  /home/E/apex/setup.py:111: UserWarning: Option --pyprof not specified. Not installing PyProf dependencies!
    warnings.warn("Option --pyprof not specified. Not installing PyProf dependencies!")

  Compiling cuda extensions with
  nvcc: NVIDIA (R) Cuda compiler driver
  Copyright (c) 2005-2023 NVIDIA Corporation
  Built on Tue_Jun_13_19:16:58_PDT_2023
  Cuda compilation tools, release 12.2, V12.2.91
  Build cuda_12.2.r12.2/compiler.32965470_0
  from /usr/local/cuda/bin

  Traceback (most recent call last):
    File "<string>", line 2, in <module>
    File "<pip-setuptools-caller>", line 34, in <module>
    File "/home/pzeng/RAG/SimCSE/apex/setup.py", line 179, in <module>
      check_cuda_torch_binary_vs_bare_metal(CUDA_HOME)
    File "/home/E/apex/setup.py", line 36, in check_cuda_torch_binary_vs_bare_metal
      raise RuntimeError(
  RuntimeError: Cuda extensions are being compiled with a version of Cuda that does not match the version used to compile Pytorch binaries.  Pytorch binaries were compiled with Cuda 12.1.
  In some cases, a minor-version mismatch will not cause later errors:  https://github.com/NVIDIA/apex/pull/323#discussion_r287021798.  You can try commenting out this check (at your own risk).
  
  Why do I get a version mismatch error even though my two major versions are the same? Later, I also directly commented out the code corresponding to the error prompt.

Reply:
I was facing the same issue. I had installed the PyTorch version with cuda 12.1 support and was using cuda 12.0 for building the apex. They don't work even though they are minor version mismatched, and after commenting out the RunTimeError message, in the later compilation stage, I ran into segmentation fault (cicc core gets dumped). I resolved the issue by reinstalling the cuda driver with version 12.1 to match my torch version.